kymatio/scattering2d/filter_bank.py

```diff
@@ -32,24 +32,28 @@ def filter_bank_real(M, N, J, L=8):
              wavelet filters.
         Notes
         -----
-        The design of the filters is optimized for the value L = 8
+        The design of the filters is optimized for the value L = 8.
     """
     filters = {}
     filters['psi'] = []
 
-
     offset_unpad = 0
     for j in range(J):
         for theta in range(L):
             psi = {}
             psi['j'] = j
             psi['theta'] = theta
-            psi_signal = morlet_2d(M, N, 0.8 * 2**j, (int(L-L/2-1)-theta) * np.pi / L, 3.0 / 4.0 * np.pi /2**j, 4.0/L, offset=offset_unpad)
+            psi_signal = morlet_2d(M, N, 0.8 * 2**j,
+                (int(L-L/2-1)-theta) * np.pi / L,
+                3.0 / 4.0 * np.pi /2**j, 4.0/L, offset=offset_unpad)
             psi_signal_fourier = fft2(psi_signal)
             for res in range(j + 1):
-                psi_signal_fourier_res = periodize_filter_fft(psi_signal_fourier, res)
-                psi[res]=torch.FloatTensor(np.stack((np.real(psi_signal_fourier_res), np.imag(psi_signal_fourier_res)), axis=2))
-                # Normalization to avoid doing it with the FFT!
+                psi_signal_fourier_res = periodize_filter_fft(
+                    psi_signal_fourier, res)
+                psi[res] = torch.FloatTensor(
+                    np.stack((np.real(psi_signal_fourier_res),
```

Comment:
@edouardoyallon re the point i was making (in a PR about plotting filters) about casting between complex and real. Here it is more serious:
This operation here is inefficient as it creates a copy. You can exploit the fact that numpy stores complex numbers as pairs of real numbers by setting a dtype and reshaping:

Check this (meaning copy and paste into an ipython):
```python
a = (np.random.randn(3, 4) + 1j* np.random.randn(3, 4)).astype('complex64')
b = a.reshape(a.shape + (1,))
b.dtype = 'float32'
print("b.shape is ", b.shape) # (3, 4, 2)
c = np.stack([np.real(a), np.imag(a)], axis=-1)
from numpy.testing import assert_array_equal
assert_array_equal(c, b)
print("a is ", a, "\n\nb is ", b, "\n\nc is ", c)
```

Don't change this in this PR. It just popped out at me as a good opportunity to explain what I was talking about a couple of days ago.